### ISSUE TITLE
我实在不明白，去年我使用F5-TTS0.3X版本，我的代码一切正常，升级了你们最新的1.0版本，根本就无法使用

### ISSUE THREAD

**Issue by mixiuemo** [SWivid/F5-TTS#1017]
### Checks

- [x] This template is only for research question, not usage problems, feature requests or bug reports.
- [x] I have thoroughly reviewed the project documentation and read the related paper(s).
- [x] I have searched for existing issues, including closed ones, no similar questions.
- [x] I am using English to submit this issue to facilitate community communication.

### Question details

没有声音，没有声音。
from flask import Flask, request, send_file, jsonify
import os
import torch
import json
from f5_tts.api import F5TTS
from f5_tts.model import DiT
from f5_tts.infer.utils_infer import load_model
import logging
import soundfile as sf
import numpy as np

# 设置日志
logging.basicConfig(level=logging.DEBUG)
logger = logging.getLogger(__name__)

app = Flask(__name__)

OUTPUT_FOLDER = './audioTTS'
app.config['OUTPUT_FOLDER'] = OUTPUT_FOLDER

if not os.path.exists(OUTPUT_FOLDER):
    os.makedirs(OUTPUT_FOLDER)

def normalize_audio(wav_data):
    """归一化音频数据"""
    if isinstance(wav_data, torch.Tensor):
        wav_data = wav_data.cpu().numpy()
    if wav_data.max() == wav_data.min():
        logger.warning("音频数据异常，尝试修复...")
        return np.zeros_like(wav_data)
    return wav_data

def check_audio(file_path):
    """检查音频文件"""
    try:
        data, samplerate = sf.read(file_path)
        logger.info(f"音频文件 {file_path} 信息:")
        logger.info(f"采样率: {samplerate}")
        logger.info(f"数据形状: {data.shape}")
        logger.info(f"数据类型: {data.dtype}")
        logger.info(f"最大值: {data.max()}")
        logger.info(f"最小值: {data.min()}")
        return True
    except Exception as e:
        logger.error(f"音频文件检查失败: {str(e)}")
        return False

def load_f5tts():
    try:
        ckpt_path = "F:\\TTSCode\\F5-TTS\\ckpts\\model_1250000.safetensors"
        vocab_path = "F:\\TTSCode\\F5-TTS\\ckpts\\vocab.txt"
        
        logger.info(f"正在加载模型: {ckpt_path}")
        logger.info(f"使用词表: {vocab_path}")
        
        model_cfg = dict(
            dim=1024, 
            depth=22, 
            heads=16, 
            ff_mult=2, 
            text_dim=512, 
            conv_layers=4,
            text_mask_padding=False, 
            pe_attn_head=1 
        )
        logger.info(f"模型配置: {model_cfg}")
        
        device = "cuda" if torch.cuda.is_available() else "cpu"
        logger.info(f"使用设备: {device}")
        
        model = load_model(
            DiT, 
            model_cfg, 
            ckpt_path, 
            vocab_file=vocab_path,
            mel_spec_type="vocos",
            device=device
        )
        logger.info("模型加载成功")
        return model
    except Exception as e:
        logger.error(f"模型加载失败: {str(e)}")
        raise

# 加载配置文件
with open('./ttsConfig.json', 'r', encoding='utf-8') as f:
    tts_config = json.load(f)
    logger.info(f"加载配置文件: {tts_config}")

f5tts = F5TTS()
f5tts.model = load_f5tts()

@app.route('/process', methods=['POST'])
def process_data():
    try:
        ref_text = tts_config.get("ref_text")  
        ref_file = tts_config.get("ref_file")
        
        logger.info(f"参考文本: {ref_text}")
        logger.info(f"参考音频: {ref_file}")

        if not ref_text or not ref_file or not os.path.exists(ref_file):
            return jsonify({"error": "请检查配置文件内容，确保提供了有效的 ref_text 和 ref_file"}), 500

        # 检查参考音频
        if not check_audio(ref_file):
            return jsonify({"error": "参考音频文件无效"}), 500

        gen_text = request.form.get('gen_text')
        logger.info(f"生成文本: {gen_text}")

        if not gen_text:
            return jsonify({"error": "请提供合成文本"}), 400

        file_uuid = request.form.get('file_uuid')
        if not file_uuid:
            return jsonify({"error": "请提供有效的 file_uuid"}), 400

        output_wave_path = os.path.join(app.config['OUTPUT_FOLDER'], f"{file_uuid}.wav")
        logger.info(f"输出文件路径: {output_wave_path}")

        # 调用 F5TTS 生成语音
        wav, sr, spect = f5tts.infer(
            ref_file=ref_file,
            ref_text=ref_text,
            gen_text=gen_text,
            file_wave=None  # 先不保存文件
        )
        
        # 处理生成的音频数据
        wav_data = normalize_audio(wav)
        
        # 手动保存音频文件
        sf.write(output_wave_path, wav_data, sr)
        
        logger.info(f"音频生成完成: 采样率={sr}")
        
        # 检查生成的音频
        if not check_audio(output_wave_path):
            return jsonify({"error": "生成的音频文件无效"}), 500

        return send_file(output_wave_path, mimetype='audio/wav')
        
    except Exception as e:
        logger.error(f"处理失败: {str(e)}", exc_info=True)
        return jsonify({"error": f"F5TTS inference failed: {str(e)}"}), 500

if __name__ == '__main__':
    app.run(debug=True)  # 开启debug模式

**Comment by SWivid** [comment]
![Image](https://github.com/user-attachments/assets/e77709c6-d273-4d99-8aef-67979b5c512d)
所以你是想用v0还是v1呢

**Comment by SWivid** [comment]
> This template is only for research question, not usage problems, feature requests or bug reports.

先看下readme把设置搞对吧
如果按照你的model_cfg是想用v0，那不升级不就行了
如果按1250000的ckpt推测你想用v1，那为什么要用v0的cfg

**Comment by mixiuemo** [comment]
先生，我目前回退了 一样不可以，我不太明白，为什么音频么有声音，(f5-tts) F:\TTSCode\F5-TTS>python flask_api.py
Building prefix dict from the default dictionary ...
Loading model from cache C:\Windows\TEMP\jieba.cache
Loading model cost 0.513 seconds.
Prefix dict has been built successfully.
Word segmentation module jieba initialized.

Download Vocos from huggingface charactr/vocos-mel-24khz

vocab :  F:\Conda\envs\f5-tts\lib\site-packages\f5_tts\infer\examples\vocab.txt
token :  custom
model :  C:\Users\XDG\.cache\huggingface\hub\models--SWivid--F5-TTS\snapshots\84e5a410d9cead4de2f847e7c9369a6440bdfaca\F5TTS_Base\model_1200000.safetensors

 * Serving Flask app 'flask_api'
 * Debug mode: off
WARNING: This is a development server. Do not use it in a production deployment. Use a production WSGI server instead.
 * Running on http://127.0.0.1:5000
Press CTRL+C to quit
Converting audio...
Using custom reference text...

Generating audio in 1 batches...
  0%|                                                                                            | 0/1 [00:00<?, ?it/s]F:\Conda\envs\f5-tts\lib\site-packages\f5_tts\model\modules.py:436: UserWarning: 1Torch was not compiled with flash attention. (Triggered internally at C:\actions-runner\_work\pytorch\pytorch\builder\windows\pytorch\aten\src\ATen\native\transformers\cuda\sdp_utils.cpp:555.)
  x = F.scaled_dot_product_attention(query, key, value, attn_mask=attn_mask, dropout_p=0.0, is_causal=False)
100%|███████████████████████████████████████████████████████████████████████████████████| 1/1 [02:25<00:00, 145.04s/it]
127.0.0.1 - - [06/May/2025 12:14:37] "POST /process HTTP/1.1" 200


from flask import Flask, request, send_file, jsonify
import os
import json
from f5_tts.api import F5TTS

app = Flask(__name__)

OUTPUT_FOLDER = './audioTTS'
app.config['OUTPUT_FOLDER'] = OUTPUT_FOLDER

if not os.path.exists(OUTPUT_FOLDER):
    os.makedirs(OUTPUT_FOLDER)
with open('./ttsConfig.json', 'r', encoding='utf-8') as f:
    tts_config = json.load(f)

f5tts = F5TTS()

@app.route('/process', methods=['POST'])
def process_data():
    ref_text = tts_config.get("ref_text")  
    ref_file = tts_config.get("ref_file")  

    # 如果配置文件中没有这些内容，返回错误
    if not ref_text or not ref_file or not os.path.exists(ref_file):
        return jsonify({"error": "请检查配置文件内容，确保提供了有效的 ref_text 和 ref_file"}), 500

    # 获取客户端传递的 gen_text 参数
    gen_text = request.form.get('gen_text')

    if not gen_text:
        return jsonify({"error": "请提供合成文本"}), 400

    # 获取客户端传递的 UUID 参数
    file_uuid = request.form.get('file_uuid')
    
    if not file_uuid:
        return jsonify({"error": "请提供有效的 file_uuid"}), 400

    # 使用传递的 UUID 作为文件名
    output_wave_path = os.path.join(app.config['OUTPUT_FOLDER'], f"{file_uuid}.wav")

    try:
        # 调用 F5TTS 生成语音
        wav, sr, spect = f5tts.infer(
            ref_file=ref_file,  # 使用配置文件中的 ref_file
            ref_text=ref_text,  # 使用配置文件中的 ref_text
            gen_text=gen_text,  # 使用客户端传递的 gen_text
            file_wave=output_wave_path,
        )
    except Exception as e:
        return jsonify({"error": f"F5TTS inference failed: {str(e)}"}), 500

    # 返回生成的音频文件
    return send_file(output_wave_path, mimetype='audio/wav')

if __name__ == '__main__':
    app.run(debug=False)

**Comment by SWivid** [comment]
使用第三方code建议自行debug适配，我们只保证本仓库代码完全可运行

**Comment by mixiuemo** [comment]
先生，我找到问题了，我本地笔记本使用的是1650 4G显卡，生成出来的没有声音，在3090和4090机器上是可以的，不管怎么说，谢谢你的回复。希望这个项目越来越好

**Comment by SWivid** [comment]
可以参考 #356 设备兼容性问题的话0.3x版本应该也一样的
